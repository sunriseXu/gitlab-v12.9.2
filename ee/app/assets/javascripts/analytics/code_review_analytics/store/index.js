import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import mutations from './mutations';
import state from './state';

Vue.use(Vuex);

const createStore = () =>
  new Vuex.Store({
    state: state(),
    actions,
    mutations,
  });

export default createStore();
