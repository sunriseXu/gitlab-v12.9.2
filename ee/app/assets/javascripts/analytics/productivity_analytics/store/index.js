import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import * as getters from './getters';
import * as actions from './actions';
import mutations from './mutations';
import filters from './modules/filters/index';
import charts from './modules/charts/index';
import table from './modules/table/index';

Vue.use(Vuex);

const createStore = () =>
  new Vuex.Store({
    state: state(),
    getters,
    actions,
    mutations,
    modules: {
      filters,
      charts,
      table,
    },
  });

export default createStore();
