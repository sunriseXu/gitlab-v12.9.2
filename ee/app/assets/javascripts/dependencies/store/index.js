import Vue from 'vue';
import Vuex from 'vuex';
import listModule from './modules/list';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';
import { DEPENDENCY_LIST_TYPES } from './constants';

Vue.use(Vuex);

export default () =>
  new Vuex.Store({
    modules: {
      [DEPENDENCY_LIST_TYPES.all.namespace]: listModule(),
    },
    actions,
    getters,
    mutations,
    state,
  });
