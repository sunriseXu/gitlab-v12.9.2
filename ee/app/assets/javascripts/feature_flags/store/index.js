import Vue from 'vue';
import Vuex from 'vuex';
import indexModule from './modules/index';
import newModule from './modules/new';
import editModule from './modules/edit';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      index: indexModule,
      new: newModule,
      edit: editModule,
    },
  });

export default createStore();
