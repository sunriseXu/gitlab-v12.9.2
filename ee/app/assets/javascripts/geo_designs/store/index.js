import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import mutations from './mutations';
import createState from './state';

Vue.use(Vuex);

const createStore = replicableType =>
  new Vuex.Store({
    actions,
    mutations,
    state: createState(replicableType),
  });
export default createStore;
