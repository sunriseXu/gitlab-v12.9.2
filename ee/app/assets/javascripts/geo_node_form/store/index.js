import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import mutations from './mutations';
import createState from './state';

Vue.use(Vuex);

const createStore = () =>
  new Vuex.Store({
    actions,
    mutations,
    state: createState(),
  });
export default createStore;
