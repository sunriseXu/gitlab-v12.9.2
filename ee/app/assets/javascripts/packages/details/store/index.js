import Vue from 'vue';
import Vuex from 'vuex';
import * as getters from './getters';

Vue.use(Vuex);

export default (initialState = {}) =>
  new Vuex.Store({
    getters,
    state: {
      ...initialState,
    },
  });
