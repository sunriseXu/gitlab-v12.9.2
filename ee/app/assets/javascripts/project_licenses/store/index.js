import Vue from 'vue';
import Vuex from 'vuex';

import listModule from './modules/list';
import { licenseManagementModule } from 'ee/vue_shared/license_management/store/index';
import { LICENSE_LIST } from './constants';
import { LICENSE_MANAGEMENT } from 'ee/vue_shared/license_management/store/constants';

Vue.use(Vuex);

export default () =>
  new Vuex.Store({
    modules: {
      [LICENSE_LIST]: listModule(),
      [LICENSE_MANAGEMENT]: licenseManagementModule(),
    },
  });
