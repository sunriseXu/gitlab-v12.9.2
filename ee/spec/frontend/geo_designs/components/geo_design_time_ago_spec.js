import Vuex from 'vuex';
import { createLocalVue, mount } from '@vue/test-utils';
import store from 'ee/geo_designs/store';
import GeoDesignTimeAgo from 'ee/geo_designs/components/geo_design_time_ago.vue';
import TimeAgo from '~/vue_shared/components/time_ago_tooltip.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('GeoDesignTimeAgo', () => {
  let wrapper;

  const propsData = {
    label: 'Test Label',
    dateString: '09-23-1994',
    defaultText: 'Default Text',
  };

  const createComponent = () => {
    wrapper = mount(GeoDesignTimeAgo, {
      localVue,
      store,
      propsData,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  const findGeoDesignTimeAgo = () => wrapper.find(GeoDesignTimeAgo);
  const findTimeAgo = () => findGeoDesignTimeAgo().find(TimeAgo);
  const findDefaultText = () => findGeoDesignTimeAgo().find('span');

  describe('template', () => {
    beforeEach(() => {
      createComponent();
    });

    it('renders GeoDesignTimeAgo container', () => {
      expect(findGeoDesignTimeAgo().exists()).toBe(true);
    });

    describe('when dateString exists', () => {
      describe('TimeAgo', () => {
        it('renders', () => {
          expect(findTimeAgo().exists()).toBe(true);
        });

        it('sets time prop', () => {
          expect(findTimeAgo().props().time).toBe(propsData.dateString);
        });

        it(`sets innerHTML as ${propsData.dateString}`, () => {
          expect(findTimeAgo().html()).toMatchSnapshot();
        });
      });

      it('hides DefaultText', () => {
        expect(findDefaultText().exists()).toBe(false);
      });
    });

    describe('when dateString is null', () => {
      beforeEach(() => {
        propsData.dateString = null;
        createComponent();
      });

      it('hides TimeAgo', () => {
        expect(findTimeAgo().exists()).toBe(false);
      });

      describe('DefaultText', () => {
        it('renders', () => {
          expect(findDefaultText().exists()).toBe(true);
        });

        it('sets innerHTML as props.defaultText', () => {
          expect(findDefaultText().html()).toMatchSnapshot();
        });
      });
    });
  });
});
