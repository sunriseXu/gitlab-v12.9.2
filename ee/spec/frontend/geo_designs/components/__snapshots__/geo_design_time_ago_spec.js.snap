// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoDesignTimeAgo template when dateString exists TimeAgo sets innerHTML as 09-23-1994 1`] = `"<time title=\\"Sep 23, 1994 12:00am GMT+0000\\" class=\\"\\">25 years ago</time>"`;

exports[`GeoDesignTimeAgo template when dateString is null DefaultText sets innerHTML as props.defaultText 1`] = `"<span>Default Text</span>"`;
